cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
I'm a little uncomfortable with that idea, because if the stack file contains multiple images, then it's possibly too much overhead each time an update is done.

Although it's easy to avoid that given the `--no-resolve-image` flag, but the some might find that frustrating UI. Anyhow, if there's no other way, then I agree that removing the check is the right thing to do.